### ISSUE TITLE
Cannot add com.google.maps.android:android-maps-utils-apklib:0.4.2:apklib to the build

### ISSUE THREAD

**Issue by lyubomyr-shaydariv** [simpligility/android-maven-plugin#718]
Hi,

Sorry if I'm asking a vague and very lengthy question, but I'm not sure if it's a real issue or a misconfiguration in my build. I'm trying to add the following [dependency](http://googlemaps.github.io/android-maps-utils/):

``` xml
<dependency>
    <groupId>com.google.maps.android</groupId>
    <artifactId>android-maps-utils-apklib</artifactId>
    <version>0.4.2</version>
    <type>apklib</type>
</dependency>
```

And this is what I get during a regular `mvn clean package` command:

```
...\app-android\target\unpacked-libs\cgma_android-maps-utils-apklib_0.4.2\AndroidManifest.xml:7: error: Error: No resource found that matches the given name (at 'value' with value '@integer/google_play_services_version').
```

The actual generated command is:

```
"C:\Program Files (x86)\Android\android-sdk\build-tools\23.0.1\aapt.exe"
    package
    --non-constant-id
    -m
    -J ...\app-android\target\generated-sources\r
    --custom-package com.google.maps.android
    -M ...\app-android\target\unpacked-libs\cgma_android-maps-utils-apklib_0.4.2\AndroidManifest.xml
    -S ...\app-android\target\unpacked-libs\cgma_android-maps-utils-apklib_0.4.2\res
    --auto-add-overlay -I "C:\Program Files (x86)\Android\android-sdk\platforms\android-15\android.jar"
    --output-text-symbols ...\app-android\target\unpacked-libs\cgma_android-maps-utils-apklib_0.4.2
```

[failing at generateRForApkLibDependency()](https://github.com/simpligility/android-maven-plugin/blob/master/src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GenerateSourcesMojo.java#L1047)

```
at com.simpligility.maven.plugins.android.CommandExecutor$Factory$DefaultCommandExecutor.executeCommand(CommandExecutor.java:248)
at com.simpligility.maven.plugins.android.phase01generatesources.GenerateSourcesMojo.generateRForApkLibDependency(GenerateSourcesMojo.java:1047)
```

If I add the missing resources manually, say:

``` xml
<aaptExtraArgs>
    <!-- this is necessary to unpack com.google.maps.android:android-maps-utils-apklib:0.4.2:apklib -->
    <aaptExtraArg>-S</aaptExtraArg>
    <aaptExtraArg>${project.build.directory}/unpacked-libs/cgag_play-services_6.5.87/res</aaptExtraArg>
</aaptExtraArgs>
```

or

``` xml
<resourceOverlayDirectories>
    <!-- this is necessary to unpack com.google.maps.android:android-maps-utils-apklib:0.4.2:apklib -->
    <resourceOverlayDirectory>${project.build.directory}/unpacked-libs/cgag_play-services_6.5.87/res</resourceOverlayDirectory>
</resourceOverlayDirectories>
```

I get a slightly different error:

```
\app-android\target\unpacked-libs\cgag_play-services_6.5.87\res\values\admob_ads_attrs.xml:10: error: Attribute "adSize" has already been defined
\app-android\target\unpacked-libs\cgag_play-services_6.5.87\res\values\admob_ads_attrs.xml:17: error: Attribute "adSizes" has already been defined
...
```

This happens because the resource gets into the generated command twice:

```
"C:\Program Files (x86)\Android\android-sdk\build-tools\23.0.1\aapt.exe"
    package
    -m
    -J ...\app-android\target\generated-sources\r
    -f
    --no-crunch -M ...\app-android\target\AndroidManifest.xml
    -S ...\app-android\src\main\android\res
    -S ...\app-android\target\unpacked-libs\cgma_android-maps-utils-apklib_0.4.2\res
    -S ...\app-android\target\unpacked-libs\cgag_play-services_6.5.87\res
    --auto-add-overlay
    -A ...\app-android\target\generated-sources\combined-assets
    -I "C:\Program Files (x86)\Android\android-sdk\platforms\android-15\android.jar"
    --output-text-symbols ...\app-android\target
    -S ...\app-android\target/unpacked-libs/cgag_play-services_6.5.87/res
```

(please note `-S ...\app-android\target/unpacked-libs/cgag_play-services_6.5.87/res` occurring two times). However, now it fails in another method, [generateR()](https://github.com/simpligility/android-maven-plugin/blob/master/src/main/java/com/simpligility/maven/plugins/android/phase01generatesources/GenerateSourcesMojo.java#L895):

```
at com.simpligility.maven.plugins.android.CommandExecutor$Factory$DefaultCommandExecutor.executeCommand(CommandExecutor.java:248)
at com.simpligility.maven.plugins.android.phase01generatesources.GenerateSourcesMojo.generateR(GenerateSourcesMojo.java:895)
```

In short, specifying no additional dependency-related parameters does not add any mention to generated commands, but specifying any of `<aaptExtraArgs>` or `<resourceOverlayDirectories>` causes the dependency resources to occur twice.

Could you please confirm whether it works as expected, or I just do it totally wrong? Thank you!
